app/controllers/reservations_controller.rb

```diff
@@ -76,6 +98,18 @@ def create_reservation
     end
   end
 
+  def destroy_reservation(reservation)
+    if @reservation.status == 'Pending'
+      if reservation.destroy
+        render json: { message: 'reservation deleted' }, status: :ok
+      else
+        render json: { error: 'Reservation can\'t be destroyed' }, status: :not_implemented
+      end
+    else
+      render json: { error: 'You can not delete this reservation' }, status: :unauthorized
+    end
+  end
```

Comment:
[Optional] I like the idea of exporting code into multiple methods to make them reusable, but this method has no use case except the one in the above, so don't you think it's better to merge them back?

Reply:
@aimalamiri  Done